cilium-cli/sysdump/sysdump.go

```diff
@@ -3294,6 +3342,119 @@ func InitSysdumpFlags(cmd *cobra.Command, options *Options, optionPrefix string,
 	hooks.AddSysdumpFlags(cmd.Flags())
 }
 
+// formatMetricsAsTable formats the raw metrics JSON into a table format like kubectl top nodes
+func (c *Collector) formatNodeMetricsAsTable(rawMetrics string) (string, error) {
+	// Parse the metrics JSON
+	var nodeMetrics metricsapi.NodeMetricsList
+	if err := json.Unmarshal([]byte(rawMetrics), &nodeMetrics); err != nil {
+		return "", fmt.Errorf("failed to parse metrics JSON: %w", err)
+	}
+
+	// Get node information to calculate percentages
+	nodes, err := c.Client.ListNodes(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to get nodes: %w", err)
+	}
+
+	// Create a map of node names to their capacity
+	nodeCapacities := make(map[string]corev1.ResourceList)
+	for _, node := range nodes.Items {
+		nodeCapacities[node.Name] = node.Status.Capacity
+	}
+
+	var sb strings.Builder
+	sb.WriteString("NAME            CPU(cores)   CPU(%)   MEMORY(bytes)   MEMORY(%)\n")
+
+	for _, metric := range nodeMetrics.Items {
+		name := metric.Name
+		if len(name) > 15 {
+			name = name[:12] + "..."
+		}
+
+		// Get current usage
+		cpuUsage := metric.Usage[corev1.ResourceCPU]
+		memUsage := metric.Usage[corev1.ResourceMemory]
+
+		cpuMillis := cpuUsage.MilliValue()
+		memBytes := memUsage.Value()
+		memMi := float64(memBytes) / (1024 * 1024)
+
+		// Calculate percentages if we have capacity info
+		cpuPercent := "<unknown>"
+		memPercent := "<unknown>"
+
+		if capacity, exists := nodeCapacities[metric.Name]; exists {
+			if cpuCap := capacity[corev1.ResourceCPU]; !cpuCap.IsZero() {
+				cpuCapMillis := cpuCap.MilliValue()
+				cpuPct := float64(cpuMillis) / float64(cpuCapMillis) * 100
+				cpuPercent = fmt.Sprintf("%.0f%%", cpuPct)
+			}
+
+			if memCap := capacity[corev1.ResourceMemory]; !memCap.IsZero() {
+				memCapBytes := memCap.Value()
+				memPct := float64(memBytes) / float64(memCapBytes) * 100
+				memPercent = fmt.Sprintf("%.0f%%", memPct)
+			}
+		}
+
+		sb.WriteString(fmt.Sprintf("%-15s %-12s %-8s %-15s %s\n",
+			name,
+			fmt.Sprintf("%dm", cpuMillis),
+			cpuPercent,
+			fmt.Sprintf("%dMi", int64(memMi)),
+			memPercent))
+	}
+
+	return sb.String(), nil
+}
+
+// formatPodMetricsAsTable formats the raw pod metrics JSON into a table format like kubectl top pods
+func (c *Collector) formatPodMetricsAsTable(rawMetrics string) (string, error) {
+	// Parse the metrics JSON
+	var podMetrics metricsapi.PodMetricsList
+	if err := json.Unmarshal([]byte(rawMetrics), &podMetrics); err != nil {
+		return "", fmt.Errorf("failed to parse pod metrics JSON: %w", err)
+	}
+
+	var sb strings.Builder
+	sb.WriteString("NAMESPACE      NAME                                 CPU(cores)   MEMORY(bytes)\n")
+
+	for _, metric := range podMetrics.Items {
+		namespace := metric.Namespace
+		if len(namespace) > 14 {
```

Comment:
The formatting here seems to be a bit adhoc? 
 Think we would want the full info in the dumps when ns/names are a mile long.